x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsProvider.java

```diff
@@ -217,7 +217,7 @@ public void checkForLeftOverDocuments(Job job, ActionListener<Boolean> listener)
         ActionListener<MultiSearchResponse> searchResponseActionListener = new DelegatingActionListener<>(listener) {
             @Override
             public void onResponse(MultiSearchResponse response) {
-                List<SearchHit> searchHits = new ArrayList<>();
+                List<String> searchHits = new ArrayList<>();
```

Comment:
This could have been pulled into its own PR, left it here for now to keep things from being blocked again. Just a trivial logical change here to reduce the logical scope of the search hits here.